src/Compilers/CSharp/Portable/BoundTree/BoundExpression.cs

```diff
@@ -280,11 +280,18 @@ public override Symbol ExpressionSymbol
 
     internal sealed partial class BoundDeconstructionAssignmentOperator : BoundExpression
     {
-        internal class AssignmentInfo
+        internal class AssignmentStep
         {
             public BoundAssignmentOperator Assignment;
-            public BoundLValuePlaceholder LValuePlaceholder;
-            public BoundRValuePlaceholder RValuePlaceholder;
+            public BoundDeconstructValuePlaceholder OutputPlaceholder;
+            public BoundDeconstructValuePlaceholder TargetPlaceholder;
+        }
+
+        internal class DeconstructStep
+        {
+            public MethodSymbol DeconstructMemberOpt; // the deconstruct member, or null if tuple deconstruction
+            public BoundDeconstructValuePlaceholder TargetPlaceholder;
```

Comment:
I would make this BoundExpression and put Right directly as the target for the first deconstruct, removing it from the parent node.